File: Principal Diagonal | VS | Secondary Diagonal.js
function diagonal(matrix){
  let result1 = matrix.map((_,i) => matrix[i][i]).reduce((a,b) => a+b,0)
  let result2 = matrix.map((v,i) => Array.from(v.reverse())[i]).reduce((a,b) => a+b,0)
  if(result1>result2){
    return 'Principal Diagonal win!'
  }else if(result2>result1){
    return 'Secondary Diagonal win!'
  }else{
    return 'Draw!'
  }
}

File: Principal Diagonal | VS | Secondary Diagonal.py
import numpy as np
def diagonal(matrix):
    matrix_n = np.array(matrix)
    matrix_nr = np.fliplr(matrix_n)
    result1 = sum([matrix_n[i,i] for i in range(len(matrix_n))])
    result2 = sum([matrix_nr[i,i] for i in range(len(matrix_n))])
    if result1==result2:
        return "Draw!"
    elif result1>result2:
        return "Principal Diagonal win!"
    else:
        return  "Secondary Diagonal win!"
    
        
